File: hotfix.cs
/* ===== إزالة أي خلفيات/جوتّات حوالين سكشن البنر (Fallback عام) ===== */
.banner-wrap,
.banner-wrap *{
  background-color: transparent !important;
}
.banner-wrap{
  padding:0 !important; margin:0 !important; box-shadow:none !important;
}
.banner-wrap .container, .banner-wrap .container-fluid{
  padding-left:0 !important; padding-right:0 !important;
}
.banner-wrap .row{
  --bs-gutter-x:0 !important; margin-left:0 !important; margin-right:0 !important;
}

/* ===== “الدلع” + شكل البانر ===== */
:root{
  --banner-max: 1120px;
  --banner-radius: 22px;
  --banner-shadow: 0 20px 50px rgba(0,0,0,.22);
  --glow-a: rgba(168,85,247,.40);
  --glow-b: rgba(34,211,238,.40);
  --dots-offset: 10px;
}

.bannerizer{
  position:relative;
  width:min(100% - 24px, var(--banner-max)) !important;
  margin:12px auto 22px !important;
  border-radius:var(--banner-radius) !important;
  overflow:hidden !important;
  line-height:0;
  box-shadow:var(--banner-shadow);
  border:1px solid transparent;
  background:
    linear-gradient(#0000,#0000) padding-box,
    linear-gradient(90deg, var(--glow-a), var(--glow-b)) border-box;
  z-index:1;
}

/* توهّج خلفي (خلف الصورة مباشرة) */
.bannerizer::before{
  content:"";
  position:absolute; inset:-28% -18% -36% -18%;
  background:
    radial-gradient(60% 40% at 20% 0%, var(--glow-a), transparent 60%),
    radial-gradient(60% 40% at 80% 100%, var(--glow-b), transparent 60%);
  filter:blur(22px); opacity:.32; pointer-events:none; z-index:0;
}

/* الحاوية الداخلية التي نثبّت عليها الـratio (القيم يضبطها JS) */
.bannerizer .slick-list{
  --arW:128; --arH:45; /* افتراضي لحد ما الـJS يقيس */
  aspect-ratio: var(--arW) / var(--arH);
  height:auto !important; min-height:0 !important;
  border-radius:inherit !important; overflow:hidden !important;
  padding:0 !important; margin:0 !important;
  background:transparent !important; position:relative; z-index:1;
}

/* املأ الارتفاع في كل طبقات السلايد */
.bannerizer .slick-track,
.bannerizer .slick-slide,
.bannerizer .slick-slide > div,
.bannerizer .slider-item,
.bannerizer .slider-item.bg-img{
  height:100% !important;
  border-radius:inherit !important;
  overflow:hidden !important;
  /* ⚠️ متعملش background: transparent هنا عشان ما تمسحش الـBG للصورة */
}

/* IMG */
.bannerizer .slick-slide img{
  width:100% !important; height:100% !important;
  object-fit:cover !important; object-position:center !important;
  display:block !important; border-radius:inherit !important;
}

/* background-image */
.bannerizer .slider-item.bg-img{
  background-size:cover !important; background-position:center !important; background-repeat:no-repeat !important;
}

/* Ken Burns ناعم */
@keyframes kb { from{ transform:scale(1); } to{ transform:scale(1.035); } }
.bannerizer .slick-current img,
.bannerizer .slick-current .slider-item.bg-img{
  animation: kb 12s ease-in-out both;
}

/* الدوتس */
.bannerizer .slick-dots{
  position:absolute !important; left:50%; transform:translateX(-50%);
  bottom:var(--dots-offset) !important; margin:0 !important; padding:0 !important;
  background:transparent !important; z-index:2;
}
.bannerizer .slick-dots li, .bannerizer .slick-dots li button{
  background:transparent !important; border:0 !important;
}

/* اختياري: فل-بليد (يملى الشاشة عرضيًا) — فعّله بإضافة كلاس full-bleed */
.bannerizer.full-bleed{
  width:100vw !important; max-width:100vw !important;
  margin-left:calc(50% - 50vw) !important; margin-right:calc(50% - 50vw) !important;
  border-radius:16px !important;
}

File: hotfix.js
// Bannerizer — اجبر البانر ياخد نسبة أبعاد صورة كل سلايد (IMG أو BG) + تنظيف اللفّاف
var TRY_SELECTORS = [
  '.main-slider-dots-style.theme1.slick-slider',
  'section.main-slider .slick-slider',
  '.home-slider .slick-slider',
  '.slider-section .slick-slider',
  '.slick-slider'
];

function findSliders(){
  // رجّع أول مجموعة موجودة
  for (var i=0;i<TRY_SELECTORS.length;i++){
    var nodes = document.querySelectorAll(TRY_SELECTORS[i]);
    if (nodes && nodes.length) return nodes;
  }
  // فallback: أول slick-slider في الصفحة
  var any = document.querySelectorAll('.slick-slider');
  return any || [];
}

function curSlide(root){
  return root.querySelector('.slick-current') ||
         root.querySelector('.slick-active')  ||
         root.querySelector('.slick-slide');
}

function readSize(slide, cb){
  if (!slide) return;
  // IMG
  var img = slide.querySelector('img');
  if (img){
    if (img.complete && img.naturalWidth){ cb({w:img.naturalWidth, h:img.naturalHeight}); return; }
    img.addEventListener('load', function(){ cb({w:img.naturalWidth, h:img.naturalHeight}); }, {once:true});
    return;
  }
  // BG
  var el = slide.matches('.slider-item, .slider-item.bg-img') ? slide :
           slide.querySelector('.slider-item, .slider-item.bg-img');
  if (el){
    var bg = getComputedStyle(el).backgroundImage;
    var m = bg && bg.match(/url\(["']?(.+?)["']?\)/);
    if (m && m[1]){
      var im = new Image();
      im.onload = function(){ cb({w:im.naturalWidth, h:im.naturalHeight}); };
      im.src = m[1];
      if (im.complete && im.naturalWidth){ cb({w:im.naturalWidth, h:im.naturalHeight}); }
    }
  }
}

function setAspect(list, w, h){
  if (!list || !w || !h) return;
  list.style.setProperty('--arW', w);
  list.style.setProperty('--arH', h);
  // fallback للمتصفحات القديمة: حساب ارتفاع يدوي
  if (!('aspectRatio' in document.documentElement.style)){
    var ww = list.clientWidth || list.getBoundingClientRect().width;
    if (ww) list.style.height = (ww * h / w) + 'px';
  }
}

function applyRatio(root){
  var list  = root.querySelector('.slick-list');
  var slide = curSlide(root);
  if (!list || !slide) return;
  readSize(slide, function(sz){ if (sz && sz.w && sz.h) setAspect(list, sz.w, sz.h); });
}

function cleanWrapper(root){
  var wrap = root.closest('section') || root.parentElement;
  if (!wrap) return;
  wrap.classList.add('banner-wrap');
  // inline لضمان الأولوية
  wrap.style.background = 'transparent';
  wrap.style.boxShadow  = 'none';
  wrap.style.padding    = '0';
  wrap.style.margin     = '0';
  var cons = wrap.querySelectorAll('.container, .container-fluid');
  cons.forEach(function(c){ c.style.paddingLeft='0'; c.style.paddingRight='0'; });
  var rows = wrap.querySelectorAll('.row');
  rows.forEach(function(r){
    r.style.marginLeft='0'; r.style.marginRight='0';
    r.style.setProperty('--bs-gutter-x','0');
  });
}

function initOne(root){
  if (root.dataset.bannerized) return;
  root.dataset.bannerized = '1';
  root.classList.add('bannerizer');

  cleanWrapper(root);     // اشطب الخلفيات/الجوتّات حوالين السلايدر

  // محاولات أوليّة لحد ما الصور/السلايدر يجهزوا
  var tries = 36, iv = setInterval(function(){
    applyRatio(root);
    if (--tries <= 0) clearInterval(iv);
  }, 250);

  // تحديثات طبيعية
  window.addEventListener('resize', function(){ applyRatio(root); }, {passive:true});
  window.addEventListener('load',   function(){ applyRatio(root); });

  // لو Slick/jQuery موجودين: امسك أحداثه
  var $ = window.jQuery;
  if ($ && $(root).on){
    $(root).on('init reInit setPosition afterChange', function(){ applyRatio(root); });
  }

  // مراقبة تغيّر الكلاسات/الستايل لو حصل سويتش داخلي
  try{
    new MutationObserver(function(){ applyRatio(root); })
      .observe(root, {subtree:true, attributes:true, attributeFilter:['class','style']});
  }catch(e){}
}

function initAll(){
  var sliders = findSliders();
  for (var i=0;i<sliders.length;i++) initOne(sliders[i]);
}

if (document.readyState === 'loading') document.addEventListener('DOMContentLoaded', initAll);
else initAll();

// (اختياري) فعّل الـfull-bleed بإضافة الكلاس من هنا
// setTimeout(function(){
//   var first = document.querySelector('.bannerizer');
//   if (first) first.classList.add('full-bleed');
// }, 800);
